protocol/messenger.go

```diff
@@ -1126,14 +1126,16 @@ func (m *Messenger) Init() error {
 			continue
 		}
 
-		m.allChats.Store(chat.ID, chat)
+                if (!chat.Timeline() && chat.ChatType != ChatTypeProfile) {
+                        m.allChats.Store(chat.ID, chat)
+                }
```

Comment:
I'm not familiar with the  `allChats.Store()` API, but I assume this affects which chats will be returned by the RPC API.  I think we need to be careful here because this will affect mobile as well. While I assume the timeline will be deprecated on a protocol level, desktop and mobile releases aren't in sync. 

So we should:

1. Either ensure with mobile that they won't be affected by this change in an unexpected way OR
2. Do what I originally proposed in https://github.com/status-im/status-desktop/pull/4198#discussion_r760106468.

> This means, what we'd need to do, is kind of the reverse of what is being done in step 3).
Instead of filling the gaps and add missing profile update channels, we should remove all chats in status-go that are of type Profile, so when we hit step 4), we don't actually create filters for those anymore.

As I'm reading this now, I see that this was probably misinterpreted. I didn't mean literally changing code in status-go, but removing channels from status-go via APIs called in desktop. That way we can safely remove the timeline feature in desktop without affecting mobile, nor the protocol in general.

Reply:
Added some more concrete comments here: https://github.com/status-im/status-desktop/pull/4198#discussion_r760931782